test/grouping.jl

```diff
@@ -2218,6 +2222,144 @@ end
     end
 end
 
+@testset "group ordering after select/transform" begin
+    df = DataFrame(g=[3, 1, 1, 2, 3], x=1:5)
+    gdf1 = groupby_checked(df, :g)
+    gdf2 = gdf1[[2, 3, 1]]
+    @test select(gdf1, :x) == select(gdf2, :x) == df
+    @test select(gdf1, :x, ungroup=false) == gdf1
+    @test select(gdf2, :x, ungroup=false) == gdf2
+    @test select(gdf1, ungroup=false) ==
+          groupby_checked(DataFrame(g=[3, 1, 1, 2, 3]), :g)
+    @test select(gdf2, ungroup=false) ==
+          groupby_checked(DataFrame(g=[3, 1, 1, 2, 3]), :g)[[2, 3, 1]]
+
+    gdf1′ = deepcopy(gdf1)
+    df1′ = parent(gdf1′)
+    gdf2′ = deepcopy(gdf2)
+    df2′ = parent(gdf2′)
+    @test select!(gdf1, :x, ungroup=false) == gdf1′
+    @test select!(gdf2, :x, ungroup=false) == gdf2′
+    @test select!(gdf1′, ungroup=false) ==
+          groupby_checked(DataFrame(g=[3, 1, 1, 2, 3]), :g)
+    @test select!(gdf2′, ungroup=false) ==
+          groupby_checked(DataFrame(g=[3, 1, 1, 2, 3]), :g)[[2, 3, 1]]
+    @test df1′ == DataFrame(g=[3, 1, 1, 2, 3])
+    @test df2′ == DataFrame(g=[3, 1, 1, 2, 3])
+end
+
+@testset "handling empty data frame / selectors / groupcols" begin
+    df = DataFrame(x=[], g=[])
+    gdf = groupby_checked(df, :g)
+
+    @test size(combine(gdf)) == (0, 1)
+    @test names(combine(gdf)) == ["g"]
+    # TODO: add tests for keepkeys and ungroup after deprecation
```

Comment:
Maybe write them know in a comment, as nobody will remember what this comment meant exactly. :-D

Reply:
OK